### PR TITLE
Better detection of our own auto-save SSE echoes

### PR DESCRIPTION
**Pull request by habdelra**
So the issue occurs specifically when changes are made while an auto-save is in-flight. the server response in this case collides with unsaved changes and overwrites unsaved changes. specifically SSE sent us a signal that the card changed on the server and the card instance in the browser has newer changes than the card instance on the server. however, we are unable to tell which version is the latest version and the server version clobbers the browser version. Instead of performing a comparison of the incoming card with the existing card and just accepting the incoming card from the server blindly, I'm doing a couple things:

1. whenever the card editor performs an auto-save we note the time. when we get an SSE event for updating a card we'll only honor the event if it occurs more than 5 seconds after the last card editor auto save
2. During the actual save, if there is a change to the card that is made while the save is in-flight, we will not load the server data into the card. rather, we'll wait for the next auto save that occurs  where the card doesn't change locally while the save is in flight before we load the server data into the local card.

Ideally some kind of versioning approach is probably a better way to handle this--but that seems like a much bigger effort. The downside to this approach is that out of band changes made within 5 seconds of a card editor auto save will be ignored by the card preview panel in code mode. This includes changes made in the monaco editor. Changes made in the monaco editor after 5 seconds of the last card editor auto-save will be reflected in the card preview panel in code mode. This seems like a reasonable tradeoff to make until we have a more robust versioning system.

### CODE REVIEW

File: packages/host/tests/acceptance/operator-mode-test.ts [hunk 1]
```diff
@@ -1593,7 +1593,6 @@ module('Acceptance | operator mode tests', function (hooks) {
     assert
       .dom('[data-test-card-url-bar-input]')
       .hasValue(`${testRealmURL}Pet/mango.json`);
-    assert.dom('[data-test-definition-name]').hasText('Mango');
```
Comment (habdelra) [inline packages/host/tests/acceptance/operator-mode-test.ts:1596]: I removed this assertion since it is not material to the test and it is flaky

**Review by backspace — APPROVED** [review]
I tried this is the preview deployment and don’t see the focus loss present in the staging environment.

Does this seem testable? Maybe not the focus aspect but the ignoring within the window… maybe not worth it.

Approving assuming the failures in other `host` tests are fixed (or are just flaky 😞)

**Comment by habdelra** [comment]
> I tried this is the preview deployment and don’t see the focus loss present in the staging environment.
> 
> Does this seem testable? Maybe not the focus aspect but the ignoring within the window… maybe not worth it.
> 
> Approving assuming the failures in other `host` tests are fixed (or are just flaky 😞)

i think the test waiters would make this impossible to test since all the interesting stuff happens during the test waiter's async. and also yes. we seem to be plagued by flaky realm-info-provider assertions

**Review by jurgenwerk — APPROVED** [review]
Sounds like a sufficient and simple strategy to avoid conflicts. The comments are really helpful, thanks!

**Pull request merged**